### ISSUE TITLE
What exactly does "linear search failed" mean in terms of tree quality?

### ISSUE THREAD

**Issue by jamiethompson77** [blackrim/treePL#49]
I'm receiving a lot of "linear search failed" lines in my run of a tree with 1457 tips.  What does this mean?

**Comment by kjlmaurin** [comment]
Hi! I would like to know this too, as I am having the same issue.

It would be awesome if the developers of treePL could provide a proper tutorial that would take readers through the whole process of running the _prime_ thing first, choosing the best smoothing parameter and _opt_ arguments and then running the actual analysis, like the developers of BEAST2 did. It is not easy to try to piece together how authors who published treePL analyses actually did their thing, because such details cannot figure in a Methods section, and contacting them to ask directly often fails (I am still waiting for answers, after quite some time...).

Many thanks!

**Comment by jamiethompson77** [comment]
Yeah I am struggling to know whether I'm doing the right steps at all, to be honest!  And how to check that each has worked correctly.  Do you know of any tutorials available online?  I can't find anything.

**Comment by kjlmaurin** [comment]
Nah that's the problem, there is no tutorial online. You kinda have to read in-between the lines of the Methods sections of paper that used treePL and try to patch the remaining holes with the info you might find on this forum. That's why I was asking if the developers could produce some tutorial, like the developers of BEAST2 did :)
I am planning on writing a tutorial based on my own experience once the paper I am writing using treePL is published (i.e. once my method is peer-reviewed ah ah!), but this will not happen before most likely another year, so in the meantime I am hoping that a general tutorial will pop up somewhere...

**Comment by jeniferclopes** [comment]
Thanks for the tutorial by the way ;)
